src/mlpack/tests/ann/layer/sum_reduce.cpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file tests/ann/layer/sumreduce.cpp
```

Comment:
This also will need to be add to `tests/ann/layer_test.cpp` to compile and run. :+1: